extensions/iceberg/src/main/java/io/deephaven/iceberg/location/IcebergTableLocationKey.java

```diff
@@ -0,0 +1,83 @@
+import io.deephaven.engine.table.impl.locations.local.URITableLocationKey;
+import io.deephaven.parquet.table.ParquetInstructions;
+import io.deephaven.parquet.table.location.ParquetTableLocationKey;
+import org.apache.iceberg.FileFormat;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.net.URI;
+import java.util.Map;
+
+/**
+ * {@link TableLocationKey} implementation for use with data stored in the parquet format.
+ */
+public class IcebergTableLocationKey extends URITableLocationKey {
+
+    private static final String IMPLEMENTATION_NAME = IcebergTableLocationKey.class.getSimpleName();
+
+    final FileFormat format;
+
+    final URITableLocationKey internalTableLocationKey;
+
+    /**
+     * Construct a new IcebergTableLocationKey for the supplied {@code fileUri} and {@code partitions}.
+     *
+     * @param fileUri The file that backs the keyed location.
+     * @param order Explicit ordering index, taking precedence over other fields
+     * @param partitions The table partitions enclosing the table location keyed by {@code this}. Note that if this
+     *        parameter is {@code null}, the location will be a member of no partitions. An ordered copy of the map will
+     *        be made, so the calling code is free to mutate the map after this call
+     * @param readInstructions the instructions for customizations while reading
+     */
+    public IcebergTableLocationKey(
+            final FileFormat format,
+            @NotNull final URI fileUri,
+            final int order,
+            @Nullable final Map<String, Comparable<?>> partitions,
+            @NotNull final Object readInstructions) {
+        super(fileUri, order, partitions);
+        this.format = format;
+        if (format == FileFormat.PARQUET) {
+            // This constructor will perform validation of the parquet file
+            final ParquetInstructions parquetInstructions = (ParquetInstructions) readInstructions;
+            this.internalTableLocationKey =
+                    new ParquetTableLocationKey(fileUri, order, partitions, parquetInstructions);
+        } else {
+            throw new IllegalArgumentException("Unsupported file format: " + format);
+        }
+    }
+
+    @Override
+    public String getImplementationName() {
+        return IMPLEMENTATION_NAME;
+    }
+
+    /**
+     * Returns {@code true} if a previous file reader has been created, or if one was successfully created on-demand.
+     *
+     * <p>
+     * When {@code false}, this may mean that the file:
+     * <ol>
+     * <li>does not exist, or is otherwise inaccessible</li>
+     * <li>is in the process of being written, and is not yet a valid parquet file</li>
+     * <li>is _not_ a parquet file</li>
+     * <li>is a corrupt parquet file</li>
+     * </ol>
+     *
+     *
+     * @return true if the file reader exists or was successfully created
+     */
```

Comment:
These comments are duplicate, you can probably just reference the `ParquetTableLocationKey.verifyFileReader` here. Please verify other comments too.